static/app/views/insights/pages/backend/laravelOverviewPage.tsx

```diff
@@ -636,6 +638,185 @@ function JobsWidget({query}: {query?: string}) {
   );
 }
 
+interface QueriesDiscoverQueryResponse {
+  data: Array<{
+    'avg(span.self_time)': number;
+    'project.id': string;
+    'span.description': string;
+    'span.group': string;
+    'span.op': string;
+    'sum(span.self_time)': number;
+    'time_spent_percentage()': number;
+    transaction: string;
+  }>;
+}
+
+function QueriesWidget({query}: {query?: string}) {
+  const organization = useOrganization();
+  const pageFilterChartParams = usePageFilterChartParams();
+
+  const queriesRequest = useApiQuery<QueriesDiscoverQueryResponse>(
+    [
+      `/organizations/${organization.slug}/events/`,
+      {
+        query: {
+          ...pageFilterChartParams,
+          dataset: 'spansMetrics',
+          field: [
+            'span.op',
+            'span.group',
+            'project.id',
+            'span.description',
+            'sum(span.self_time)',
+            'avg(span.self_time)',
+            'time_spent_percentage()',
+            'transaction',
+          ],
+          query: `has:span.description span.module:db ${query}`,
+          referrer: 'api.performance.generic-widget-chart.most-time-spent-db-queries',
+          sort: '-time_spent_percentage()',
+          per_page: 3,
+        },
+      },
+    ],
+    {staleTime: 0}
+  );
+
+  const timeSeriesRequest = useApiQuery<MultiSeriesEventsStats>(
+    [
+      `/organizations/${organization.slug}/events-stats/`,
+      {
+        query: {
+          ...pageFilterChartParams,
+          dataset: 'spansMetrics',
+          field: ['span.group', 'sum(span.self_time)'],
+          yAxis: ['sum(span.self_time)'],
+          query: `span.group:[${queriesRequest.data?.data.map(item => item['span.group']).join(',')}]`,
+          orderby: '-sum(span.self_time)',
+          topEvents: 3,
+        },
+      },
+    ],
+    {
+      staleTime: 0,
+      enabled: !!queriesRequest.data?.data,
+    }
+  );
+
+  const timeSeries = useMemo<TimeSeries[]>(() => {
+    if (
+      !timeSeriesRequest.data ||
+      !queriesRequest.data ||
+      'data' in timeSeriesRequest.data
+    ) {
+      return [];
+    }
+
+    const groupNameMap = new Map<string, string>(
+      queriesRequest.data.data.map(item => [item['span.group'], item['span.description']])
+    );
+
+    return Object.keys(timeSeriesRequest.data).map(key => {
+      const groupName = groupNameMap.get(key) ?? key;
+      return {
+        data:
+          timeSeriesRequest.data[key]?.data.map(([time, [value]]) => ({
+            timestamp: new Date(time * 1000).toISOString(),
+            value: value?.count!,
+          })) ?? [],
+        field: groupName,
+        meta: {
+          fields: {
+            [groupName]:
+              timeSeriesRequest.data[key]?.meta?.fields?.sum_span_self_time ?? '',
+          },
+          units: {
+            [groupName]:
+              timeSeriesRequest.data[key]?.meta?.units?.sum_span_self_time ?? '',
+          },
+        },
+      };
```

Comment:
Unfortunately this isn't explained anywhere, but you can pass the `transformAliasToInputFormat` URL query param to the `/events-stats/` endpoint, and it'll return meta in the same format as the input query. The returned meta will be

```json
{
  fields: {
   "sum(span.self_time)": "...
```

and you won't have to rename it 👍🏻